articles/dms/howto-mysql-to-azure-mysql-powershell.md

```diff
@@ -0,0 +1,657 @@
+# DEFAULT values for all the configurations is $null
+LogMessage -Message "Adding optional migration performance tuning configuration ..." -IsProcessing $true
+# Partitioning settings
+[object] $DesiredRangesCount = 4
+[object] $MaxBatchSizeKb = 4096
+[object] $MinBatchRows = $null
+# Task count settings
+[object] $PrepareDatabaseForBulkImportTaskCount = $null
+[object] $PrepareTableForBulkImportTaskCount = $null
+[object] $QueryTableDataRangeTaskCount = 8
+[object] $WriteDataRangeBatchTaskCount = 12
+# Batch cache settings
+[object] $MaxBatchCacheSizeMb = $null
+[object] $ThrottleQueryTableDataRangeTaskAtAvailableMemoryMb = $null
+[object] $ThrottleQueryTableDataRangeTaskAtBatchCount = 36
+# Performance settings
+[object] $DelayProgressUpdatesInStorageInterval = "00:00:30"
+
+function AddOptionalSetting($optionalAgentSettings, $settingName, $settingValue) {
+    # If no value specified for the setting, don't bother adding it to the input
+    if ($null -eq $settingValue) {
+        return;
+    }
+
+    # Add a new property to the JSON object to capture the setting which will be customized
+    $optionalAgentSettings | add-member -MemberType NoteProperty -Name $settingName -Value $settingValue
+}
+
+# Set any optional settings in the input based on parameters to this cmdlet
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "DesiredRangesCount" $DesiredRangesCount;
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "MaxBatchSizeKb" $MaxBatchSizeKb;
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "MinBatchRows" $MinBatchRows;
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "PrepareDatabaseForBulkImportTaskCount" $PrepareDatabaseForBulkImportTaskCount;
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "PrepareTableForBulkImportTaskCount" $PrepareTableForBulkImportTaskCount;
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "QueryTableDataRangeTaskCount" $QueryTableDataRangeTaskCount;
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "WriteDataRangeBatchTaskCount" $WriteDataRangeBatchTaskCount;
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "MaxBatchCacheSizeMb" $MaxBatchCacheSizeMb;
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "ThrottleQueryTableDataRangeTaskAtAvailableMemoryMb" $ThrottleQueryTableDataRangeTaskAtAvailableMemoryMb;
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "ThrottleQueryTableDataRangeTaskAtBatchCount" $ThrottleQueryTableDataRangeTaskAtBatchCount;
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "DelayProgressUpdatesInStorageInterval" $DelayProgressUpdatesInStorageInterval;
+```
+
+## Step 11: Running the migration task
+
+After configuration of the task, you will now execute the task and wait for the migration to complete.
```

Comment:
After configuring the input for the task, now it will be created / executed on the agent.

Reply:
Done